reference/best_practices/reusable_bundles.rst

```diff
@@ -1,4 +1,95 @@
 Reusable Bundles for Akeneo PIM community
 =========================================
 
-Coming soon!!! :)
+You're integrator and you want to share your project with the awesome Akeneo PIM community?
+Let's read our advices to make a reusable and documented bundle!
+
+
+What do you want when you are creating a reusable bundle?
+
+* **Compatibility**: fit Akeneo PIM features
+* **Maintainability**: make your code flexible and testable (and tested)
+* **Extensibility**: to allow integrators to fit your bundle with their specific needs
+* **Scalability**: know your application limits and document them
+
+
+Compatibility
+-------------
+
+Your bundle won't be reusable for some customers if you forgot some cases. But that's not a problem if they are aware of the compatibility of your bundle.
+
+Akeneo PIM storage stack comes with MySQL and hybrid MySQL/MongoDB (:doc:`/reference/technical_information/choose_database`).
+Use the ProductQueryBuilder which is an overlayer making work both storages with the same API.
+The following cookbook has been done to help you: ":doc:`/cookbook/catalog/product/query`".
+
+Two existing distributions CE/EE. The main project is `Akeneo PIM Community Edition`_ (CE) but
+we also have an `Enterprise Edition`_ (EE) which comes with a bunch of extra features dedicated to advanced needs.
+
+Many attribute types are managed (14 in CE and one more in EE). Be sure you don't forget a use case for an attribute type.
+
+Try your bundle on another catalogs. There is a multiplicity of catalog possibilities.
+Don't forget use cases with many locales, many channels and various catalog volumes.
+You can use our `data generator bundle`_ to generate a catalog.
+
+.. _Akeneo PIM Community Edition: https://github.com/akeneo/pim-community-standard
+.. _Enterprise Edition: https://www.akeneo.com/enterprise-edition/
+.. _data generator bundle: https://github.com/akeneo-labs/DataGeneratorBundle
+
+
+Maintainability
+---------------
+
+First, keep in mind the `Single Responsibility` from the five SOLID principles of Oriented Object Programming.
+It seems simple but we often forget to explode our code in many thinner classes.
+The code is easily readable and testable.
+
+You can use `PMD`_ to help you to make your code simple, flexible and easily testable.
+
+
+Now your code should be testable, so test your code! :)
+In Akeneo PIM, we use `PhpSpec`_ as unit testing tool and `Behat`_ for to test our application.
+
+Use a Continuous Integration to deploy your application, test your code and improve its quality.
+On our open-source projects, we use free CI tools `Travis`_ and `Scrutinizer`_
```

Comment:
Explain what Travis is for in few words, same for Scrutinizer so we quickly understand what is it all about. You should add (CI) right after Continuous Integration first use.